lib/error.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+/**
+ * Custom error class used in unifile
+ */
+class UnifileError extends Error {
+	static get ENOTSUP() { return 'ENOTSUP'; }
+	static get EISDIR() { return 'EISDIR'; }
+	static get EACCES() { return 'EACCES'; }
+	static get EINVAL() { return 'EINVAL'; }
+	static get ENOENT() { return 'ENOENT'; }
+	static get EIO() { return 'EIO'; }
+
+	constructor(code, message, ...params) {
+		super(message, ...params);
+		this.name = 'UnifileError';
+		this.code = code;
+		// use defineProperty in order to send the message field
+		// when using express res.send(err)
+		Object.defineProperty(this, 'message', {
```

Comment:
Why don't you use `this.message = message` ? Just to make it read-only?

Reply:
yes I think this was the idea